src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -212,4 +225,47 @@ public void verifyAccount(BillingProfileRequestModel request, AuthenticatedUserR
         }
     }
 
+    // -- profile upgrade --
+
+    // Billing profiles prior to the introduction of this service did not have sam resources behind them.
+    // This code generates sam resources for any billing profile it does not have one. It sets the
+    // stewards group as the owner of the billing profile.
+    //
+    // We assume this is being run as a Steward and that the Steward has access to any billing
+    public UpgradeResponseModel upgradeProfileResources(UpgradeModel request, AuthenticatedUserRequest user) {
+        Instant startTime = Instant.now();
+        List<BillingProfileModel> profiles = profileDao.getOldBillingProfiles();
+        List<UUID> resources = iamService.listAuthorizedResources(user, IamResourceType.SPEND_PROFILE);
```

Comment:
does this mean that this needs to be run for every single user?  Or is there an uber user that runs this and can see all the resources for everyone?

Reply:
I'll write a comment here also. In the current state, no billing profiles have sam resources, so this list will be empty.
When we upgrade a billing profile, we give stewards `owner` role. So assuming we run this as a steward, (and I'll add that check on the upgrade endpoint) we will be able to see any profiles we have already converted.

The hole here is if someone gets in and creates a billing profile with a resource _before_ we run the upgrade. We would not retrieve that one here. However, Sam would not let us create the resource in the step below. We would log an error, but otherwise, things would work. I think. I hope.